packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -552,3 +560,17 @@ defineProperty(LightningElement, 'CustomElementConstructor', {
 if (process.env.NODE_ENV !== 'production') {
     patchLightningElementPrototypeWithRestrictions(LightningElement.prototype);
 }
+
+defineProperty(LightningElement, 'shadow', {
+    get() {
+        return true;
+    },
+    set(v) {
```

Comment:
What happens if the value is updated?

```
import { LightningElement } from 'lwc';

export default class LightDOM extends LightningElement {
  constructor() {
    super();
    LightDOM.shadow = true;
  }

  static shadow = false;
}
```

I would recommend going with the most constrained approach for now by making this field immutable and throwing an error if the value is updated after it being set.